kong2tf/builder_default_terraform.go

```diff
@@ -0,0 +1,407 @@
+package kong2tf
+
+import (
+	"crypto/md5"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/kong/go-database-reconciler/pkg/file"
+	"github.com/mitchellh/hashstructure"
+)
+
+// cleanField removes all characters from the input string that are not letters, digits, underscores, and dashes.
+func cleanField(input string) string {
+	// Regular expression to match disallowed characters and replace them
+	re := regexp.MustCompile(`[^a-zA-Z0-9_-]`)
+	return re.ReplaceAllString(input, "")
+}
+
+// dashToUnderscore replaces all dashes in the input string with underscores.
+func dashToUnderscore(input string) string {
+	// Regular expression to match dashes and replace them with underscores
+	re := regexp.MustCompile(`-`)
+	return re.ReplaceAllString(input, "_")
+}
+
+var funcs = template.FuncMap{
+	"hash":             hashstructure.Hash,
+	"jsonmarshal":      json.Marshal,
+	"cleanField":       cleanField,
+	"dashToUnderscore": dashToUnderscore,
+}
+
+type DefaultTerraformBuider struct {
+	content string
+}
+
+func newDefaultTerraformBuilder() *DefaultTerraformBuider {
+	return &DefaultTerraformBuider{}
+}
+
+// Generic function that takes type T and returns map[string]any using JSON marshalling
+func toMapAny(resource any) map[string]any {
+	resourceMap := make(map[string]interface{})
+	resourceJSON, err := json.Marshal(resource)
+	if err != nil {
+		log.Fatal(err, "Failed to marshal resource")
+		return resourceMap
+	}
+	err = json.Unmarshal(resourceJSON, &resourceMap)
+	if err != nil {
+		log.Fatal(err, "Failed to unmarshal resource")
+		return resourceMap
+	}
+	return resourceMap
+}
+
+func (b *DefaultTerraformBuider) buildControlPlaneVar(content *file.Content, generateImportsForControlPlaneID *string, ignoreCredentialChanges bool) {
+	cpId := "YOUR_CONTROL_PLANE_ID"
+	if generateImportsForControlPlaneID != nil {
```

Comment:
If `generateImportsForControlPlaneID` is nil, cpId is left to "YOUR_CONTROL_PLANE_ID". I am assuming that the user is supposed to replace this string with their own CP id in the generated Terraform file. Is that assumption correct?

Reply:
Yes, that's correct. If they provide `generateImportsForControlPlaneID` we populate it for them